### PR TITLE
Es/notify restricted domain

### PR DESCRIPTION
**Pull request by esoergel**
##### SUMMARY
See Product Description.

##### PRODUCT DESCRIPTION
Make it clear to superusers that access is intentionally blocked.  I wrote this a couple months ago, then I think ran into issues testing it out and kinda forgot about it.  This came up again just now, so I figured I'd push it over the hump.

image

### CODE REVIEW

File: corehq/apps/domain/decorators.py [hunk 1]
```diff
@@ -95,52 +95,53 @@ def call_view(): return view_func(req, domain_name, *args, **kwargs)
...
-                    )
-                else:
-                    return call_view()
-
-            elif (
-                _page_is_whitelist(req.path, domain_name) or
-                not domain_obj.restrict_superusers
-            ) and user.is_superuser:
-                # superusers can circumvent domain permissions.
+        if not (user.is_authenticated and user.is_active):
```
Comment (esoergel) [inline corehq/apps/domain/decorators.py]: I recommend this one with [whitespace off](https://github.com/dimagi/commcare-hq/pull/26292/commits/51bdcb8d470cfeecf66def98aaff75b3021b5d69?w=1)

**Review by kaapstorm — APPROVED** [review]
Nice!

File: corehq/apps/domain/decorators.py [hunk 2]
```diff
@@ -43,12 +39,10 @@
     formplayer_auth,
     get_username_and_password_from_request,
 )
-# CCHQ imports
 from corehq.apps.domain.models import Domain, DomainAuditRecordEntry
 from corehq.apps.domain.utils import normalize_domain_name
 from corehq.apps.hqwebapp.signals import clear_login_attempts
 from corehq.apps.users.models import CouchUser
-########################################################################################################
```
Comment (kaapstorm) [inline corehq/apps/domain/decorators.py]: I vaguely remember reading a rant from Linus Torvalds about stuff like this. He did not love it, and he made his feelings abundantly clear, in his special way. :upside_down_face:

**Comment by orangejenny** [comment]
💯

**Pull request merged**